### PR TITLE
Extract wow / wow&

### PR DESCRIPTION
**Pull request by AnEmortalKid**
more extraction of parsing logic into handle functions for `wow` and `wow&`. Also split them up and added some tests (b4 splitting just to make sure thing still worked)

### CODE REVIEW

**Comment by vpzomtrrfrt** [comment]
Why split them?  The code seems nearly identical

**Comment by AnEmortalKid** [comment]
> Why split them? The code seems nearly identical

good point, that original if statement was a bit confusing so i split them up. I have an idea on how to merge them now, so I'll give that a go and we can see if that seems nicer!

**Comment by AnEmortalKid** [comment]
@vpzomtrrfrt does this look better than the split version?

File: lib/parser.js [hunk 1]
```diff
@@ -77,6 +77,17 @@ function expectToken(expectedStart, tokens)
   }
 }
 
+/**
+ * Raises an error if the given tokens do not start with either of the desired states
+ */
+function expectEitherToken(expectedStart, alternateStart, tokens)
```
Comment (vpzomtrrfrt) [inline lib/parser.js]: Maybe this should be extended to any number of options?

File: lib/parser.js [hunk 2]
```diff
@@ -383,6 +394,44 @@ function controlFlowParser(keyStart, keys)
   return statement;
 }
 
+/**
+ * Handles the ending of statements:
+ * wow [return vals]
+ * wow& [return vals]
+ */
+function handleWow_WowAnd(keys)
```
Comment (vpzomtrrfrt) [inline lib/parser.js]: The underscore seems strange here.  I think it would be okay to just call it handleWow

**Pull request merged**